app/routes/submissions.rb

```diff
@@ -15,8 +15,15 @@ class Submissions < Core
 
         title("%s by %s in %s" % [submission.problem.name, submission.user.username, submission.problem.language])
 
+        comment_history = if current_user.guest? || !$flipper[:comment_history].enabled?(current_user)
```

Comment:
May I suggest an structure like `lib/flipper/comment_history.rb` ?

This class would be initialized with the current user and flipper and it would be easy to test. Another benefit is that it would be easy to remove this feature in the future. Future toggle features would also be easily spotted since they would all be located in the same folder.